lib/block-supports/behaviors.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Behaviors block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers the behaviors block attribute for block types that support it.
+ * And add the needed hooks to add those behaviors.
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function gutenberg_register_behaviors_support( $block_type ) {
+	$has_behaviors_support = block_has_support( $block_type, array( 'behaviors' ), false );
+	if ( ! $has_behaviors_support ) {
+		return;
+	}
+
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	$block_type->attributes['behaviors'] = array(
+		'type' => 'object',
+	);
+
+	// If it supports the lightbox behavior, add the hook to that block.
+	// In the future, this should be a loop with all the behaviors.
+	$has_lightbox_support = block_has_support( $block_type, array( 'behaviors', 'lightbox' ), false );
+	if ( $has_lightbox_support ) {
+		add_filter( 'render_block_' . $block_type->name, 'gutenberg_blocks_supports_add_lightbox', 15, 2 );
```

Comment:
Why are we using priority 15 instead of 10? We probably want to add a comment explaining that.

I can see that we'd want it to run later than the other block supports, but I figured that's why you were using the `render_block_{block_name}` instead of `render_block` filter.

Reply:
I thought that there could be external plugins using the `render_block_{block_name}` filter to modify the image as the duotone is doing, and I believe this hook should run after the default ones.

I'll add a comment for that. Thank you! 🙂